### PR TITLE
Feature/p22 37 start page

### PR DESCRIPTION
**Pull request by FilipK7**

### CODE REVIEW

File: tennis-scheduler-web-app/src/api/TennisCourtApi.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import axios from "axios";
+
+let baseUrl = "http://localhost:8081/tennis-courts/";
```
Comment (markoacimovic) [inline tennis-scheduler-web-app/src/api/TennisCourtApi.js]: You could extract URL in .env file, with that URL will be only in one place

File: tennis-scheduler-web-app/src/components/TennisCourts.jsx [hunk 2]
```diff
@@ -1,9 +1,41 @@
...
-    )
-}
-
+  useEffect(() => {
+    async function getCourts() {
+      getTennisCourts().then((data) => {
+        setTennisCourts(data.data);
+        console.log(data.data);
+      });
+    }
```
Comment (markoacimovic) [inline tennis-scheduler-web-app/src/components/TennisCourts.jsx]: Extract this as a new function

File: tennis-scheduler-web-app/src/components/TennisCourtInfo.jsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
+function TennisCourtInfo({ image, name, surfaceType, description }) {
```
Comment (markoacimovic) [inline tennis-scheduler-web-app/src/components/TennisCourtInfo.jsx]: Try to be consistent, use only arrow or only normal functions

File: tennis-scheduler-web-app/src/components/Navbar.jsx [hunk 4]
```diff
@@ -1,21 +1,25 @@
...
+import Nav from "react-bootstrap/Nav";
+import Navbar from "react-bootstrap/Navbar";
+import { Link } from "react-router-dom";
 
 export const NavbarStart = () => {
   return (
-    <Navbar bg="light" expand="lg" fixed='top'>
+    <Navbar bg="light" expand="lg" fixed="top">
       <Container>
         <Navbar.Brand href="/">Tennis scheduler </Navbar.Brand>
```
Comment (markoacimovic) [inline tennis-scheduler-web-app/src/components/Navbar.jsx]: Try to use Link here, to be more consistent

File: tennis-scheduler-web-app/src/components/TennisCourts.jsx [hunk 5]
```diff
@@ -1,9 +1,41 @@
...
-    return (
-        <> </>
-    )
-}
-
+  useEffect(() => {
+    async function getCourts() {
+      getTennisCourts().then((data) => {
+        setTennisCourts(data.data);
+        console.log(data.data);
```
Comment (cyberpoetry17) [inline tennis-scheduler-web-app/src/components/TennisCourts.jsx]: Remove unnecessary logs.

File: tennis-scheduler-web-app/src/components/TennisCourts.jsx [hunk 6]
```diff
@@ -1,9 +1,41 @@
...
+      });
+    }
+    getCourts();
```
Comment (cyberpoetry17) [inline tennis-scheduler-web-app/src/components/TennisCourts.jsx]: What happens if request fails?

File: tennis-scheduler-web-app/src/components/TennisCourts.jsx [hunk 7]
```diff
@@ -1,9 +1,41 @@
-import React from "react"
+import React, { useState } from "react";
+import { useEffect } from "react";
```
Comment (cyberpoetry17) [inline tennis-scheduler-web-app/src/components/TennisCourts.jsx]: Can be in single line:
import React, { useState, useEffect } ....

File: tennis-scheduler-web-app/.env [hunk 8]
```diff
@@ -0,0 +1 @@
+REACT_APP_TENNIS_COURTS=http://localhost:8081/tennis-courts/
```
Comment (markoacimovic) [inline tennis-scheduler-web-app/.env]: Its better in .env only to be http://localhost:8081/ and later in files for sending request you add what you need. With that everyone could use it for their requests

File: tennis-scheduler-web-app/src/components/TennisCourts.jsx [hunk 9]
```diff
@@ -1,9 +1,44 @@
...
+    getCourts();
+  }, []);
 
+  async function getCourts() {
+    getTennisCourts()
+      .then((data) => {
+        setTennisCourts(data.data);
+      })
+      .catch((err) => {
+        console.log(err);
```
Comment (markoacimovic) [inline tennis-scheduler-web-app/src/components/TennisCourts.jsx]: It would be nice to show user that there was an error, on screen not in console

Comment (cyberpoetry17) [reply tennis-scheduler-web-app/src/components/TennisCourts.jsx]: Also, use full length naming -> error

File: tennis-scheduler-web-app/src/components/TennisCourts.jsx [hunk 10]
```diff
@@ -1,9 +1,50 @@
...
+  async function getCourts() {
+    getTennisCourts()
+      .then((data) => {
+        setTennisCourts(data.data);
+        setErrorTennisCourt("");
+        setIsShow(false);
+      })
+      .catch((error) => {
+        setErrorTennisCourt("Couldn't load tennis courts.");
+        console.log(error);
```
Comment (cyberpoetry17) [inline tennis-scheduler-web-app/src/components/TennisCourts.jsx:24]: No need for console logging.

**Review by cyberpoetry17 — APPROVED** [review]

File: tennis-scheduler-web-app/src/components/TennisCourts.jsx [hunk 11]
```diff
@@ -1,9 +1,50 @@
...
+        setIsShow(true);
+      });
+  }
 
+  return (
+    <>
```
Comment (markoacimovic) [inline tennis-scheduler-web-app/src/components/TennisCourts.jsx]: No need for empty <> </>

**Review by markoacimovic — APPROVED** [review]

**Pull request merged**